packages/ir-sdk/fern/apis/ir-types-latest/definition/http.yml

```diff
@@ -198,6 +199,18 @@ types:
     properties:
       error: errors.DeclaredErrorName
 
+  Pagination:
+    docs: |
+      If set, the endpoint will be generated with auto-pagination features.
+
+      The page must be defined as a query parameter included in the request,
+      whereas the next page and results are resolved from properties defined
+      on the response.
+    properties:
+      page: QueryParameter
+      next: types.ObjectProperty
+      results: types.ObjectProperty
```

Comment:
Thinking this through some more -

This might not actually be enough information for the generators to deduce where the property is defined. This works fine if the property is a top-level property within the response object like so:

```yaml
types:
  Response:
    properties:
      next: string
      results: list<User>
```

However, if the property is nested within another property like so:

```yaml
types:
  PageInfo:
    properties:
      next: string

  Response:
    properties:
      pageInfo: PageInfo
      results: list<User>
```

In this case, we still want to support a pagination configuration that permits nested properties, i.e.

```yaml
pagination:
  page: $request.cursor
  next: $response.pageInfo.next
  results: $response.results
```

--

We'll either need to leave in some breadcrumbs for the generator to identify where the `next` and `results` are located, or something else similar. Something like,

```yaml
PageProperty:
  properties:
    breadcrumbs: list<commons.Name> # Empty if the response defines the property at the top-level.
    property: types.ObjectProperty
```

Reply:
I went ahead went this change for now.

Reply:
Discussed offline (as shown by the image above). In the future we'll reduce the size of the IR by using IDs wherever possible. For now, this is fine as-is.